community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StoreNodeRelationshipCursor.java

```diff
@@ -86,8 +86,15 @@ public StoreNodeRelationshipCursor init( boolean isDense,
 
         if ( isDense && firstRelId != Record.NO_NEXT_RELATIONSHIP.intValue())
         {
-            groupRecord = groupStore.getRecord( firstRelId, groupRecordInstance );
-            relationshipId = nextChainStart();
+            try
+            {
+                groupRecord = groupStore.getRecord( firstRelId, groupRecordInstance );
+                relationshipId = nextChainStart();
+            }
+            catch ( InvalidRecordException e )
```

Comment:
We should not rely on exceptions. But that store doesn't expose any better `getRecord` method. I'm working on a unified getRecord thing for all stores, so we can do this later.